File: java/structure/linear/stack/StackOfLinkedList.java
package structure.linear.stack;

public class StackOfLinkedList {
    private ListNode _head = new ListNode(0);

    public boolean isEmpty() {
        return null == _head.next;
    }

    public void push(int value) {
        ListNode node = new ListNode(value);
        node.next = _head.next;
        _head.next = node;
    }

    public int pop() {
        if (null != _head.next) {
            ListNode node = _head.next;
            _head.next = node.next;
            return node.value;
        }
        return Integer.MIN_VALUE;
    }

    public void clear() {
        _head.next = null;
    }

    private class ListNode {
        public int value = 0;
        public ListNode next = null;

        public ListNode(int value) {
            this.value = value;
        }
    }
}

File: py/structure/linear/stack/StackOfLinkedList.py
#!/usr/bin/env python
# -*- coding:gbk -*-

import sys

class StackOfLinkedList:
    def __init__(self):
        self.__head = self.__ListNode(0)

    def isEmpty(self) -> bool:
        return not self.__head.next

    def push(self, value: int) -> None:
        node = self.__ListNode(value)
        node.next, self.__head.next = self.__head.next, node

    def pop(self) -> int:
        if self.__head.next:
            node = self.__head.next
            self.__head.next = node.next
            return node.value
        return -sys.maxsize

    def clear(self) -> None:
        self.__head.next = None

    class __ListNode:
        def __init__(self, value: int):
            self.value, self.next = value, None
